csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,572 @@
+        convention_name_matches = [
+            standard_name
+            for standard_name, options in CONVENTION_NAMES.items()
+            if column_name.lower() in options
+        ]
+        column_type = (
+            convention_name_matches[0] if convention_name_matches else "dimension"
+        )
+        # cols[i] = {'index': i, 'column_name': column_name, 'qb_type': column_type }
+        column_types.append(column_type)
+
+    for i, column_type in enumerate(column_types):
+        # Build a dict of fields for the identified column types' data-class
+        column_name = data.columns[i]
+        column_data = data.T.iloc[i]
+
+        if column_type == "dimension":
+            column_dict = {"type": column_type, "new": True, "value": column_name}
+
+        elif column_type == "observations":
+            column_dict = {
+                "type": column_type,
+                "value": column_name,
+                "datatype": "decimal",
+            }
+
+        elif column_type in ["measures", "units"]:
+            column_dict = {"type": column_type, "new": True}
+        elif column_type == "attribute":
+            # Note: attribute type columns are currently not supported
+            raise ValueError(
+                "Column type 'Attribute' is not supported when using csv naming convention"
+            )
+
+        else:
+            raise ValueError(f"Column type '{column_type}' is not supported.")
+
+        # As all columns will be new in this scenario get a pd.Series for the column
+
+        try:
+            qb_column = map_column_to_qb_component(
+                column_title=column_name,
+                column=column_dict,
+                data=column_data.astype("category"),
+                info_json_parent_dir=None,
+            )
+            cube_columns.append(qb_column)
+
+        except Exception as err:
+            log.error(f"{type(err)} exception raised because: {repr(err)}")
+
+    cube_metadata = _metadata_from_dict(
+        {"id": str(csv_path).upper(), "title": csv_path.name.upper().replace("_", " ")}
+    )
+    # TODO - Create the new Cube
+    cube = Cube(metadata=cube_metadata, data=data, columns=cube_columns)
+    # validation_errors = cube.validate()
+    validation_errors = cube.pydantic_validation()
+
+    return cube, validation_errors
+
+
+def get_cube_from_info_json(
+        config_path: Path,
+        data: pd.DataFrame,
+        cube_id: Optional[str] = None,
+        log: logging.RootLogger = None
+) -> Tuple[QbCube, List[ValidationError]]:
+    """
+    Generates a Cube structure from a config.json input.
+    :return: tuple of cube and json schema errors (if any)
+    """
+
+    config = _load_resource(config_path)
+    schema_path = Path("..", "..", "..", "..", "csvcubed", "schema", "cube-config-schema.json").resolve()
+    schema = _load_resource(schema_path)
+    try:
+        errors = validate_dict_against_schema(value=config, schema=schema, log=log)
```

Comment:
I think I prefer the approach where if there are schema validation errors, we present them to the user but allow the process to complete as best it can.

There are often schema validation errors which occur because someone is using a dimension URI which hasn't been defined in our list of known dimensions; the dimension URI may still be valid but we just haven't included it yet in our JSON schema.